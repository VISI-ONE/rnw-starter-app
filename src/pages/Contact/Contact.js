import React, { Component } from 'react';
import { StyleSheet, View, Text } from 'react-native';

export class Contact extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text>Contact Page</Text>
      </View>
    );
  }
}

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  }
});

export default Contact;
