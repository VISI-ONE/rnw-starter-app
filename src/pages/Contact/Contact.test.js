import React from 'react';
import { View, Text, Platform } from 'react-native';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Contact, { styles } from './Contact';

describe('Contact page', () => {
  it('render properly - ' + Platform.OS, () => {
    const tree = renderer.create(<Contact />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render the proper structure', () => {
    const tree = shallow(<Contact />);
    expect(
      tree.contains(
        <View style={styles.container}>
          <Text>Contact Page</Text>
        </View>
      )
    ).toBe(true);
  });
});
