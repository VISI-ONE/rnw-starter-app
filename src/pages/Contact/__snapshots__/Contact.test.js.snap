// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact page render properly - ios 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Contact Page
  </Text>
</View>
`;
